fhir-path/src/main/java/com/ibm/fhir/path/tools/Main.java

```diff
@@ -0,0 +1,386 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.path.tools;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.ibm.fhir.model.format.Format;
+import com.ibm.fhir.model.parser.FHIRParser;
+import com.ibm.fhir.model.parser.exception.FHIRParserException;
+import com.ibm.fhir.model.resource.Resource;
+import com.ibm.fhir.path.FHIRPathBooleanValue;
+import com.ibm.fhir.path.FHIRPathDateTimeValue;
+import com.ibm.fhir.path.FHIRPathDateValue;
+import com.ibm.fhir.path.FHIRPathDecimalValue;
+import com.ibm.fhir.path.FHIRPathElementNode;
+import com.ibm.fhir.path.FHIRPathIntegerValue;
+import com.ibm.fhir.path.FHIRPathNode;
+import com.ibm.fhir.path.FHIRPathQuantityNode;
+import com.ibm.fhir.path.FHIRPathQuantityValue;
+import com.ibm.fhir.path.FHIRPathResourceNode;
+import com.ibm.fhir.path.FHIRPathStringValue;
+import com.ibm.fhir.path.FHIRPathTimeValue;
+import com.ibm.fhir.path.FHIRPathTypeInfoNode;
+import com.ibm.fhir.path.evaluator.FHIRPathEvaluator;
+import com.ibm.fhir.path.exception.FHIRPathException;
+import com.ibm.fhir.path.visitor.FHIRPathNodeVisitor;
+
+/**
+ * Processes the resource from one of the entry points, into an in memory resource
+ * a FHIR Path is executed over it, and the output is put out to the screen.
+ *
+ * The types of entry points are:
+ * a file
+ * a string (raw command line arg); or
+ * stdin (so we can pipe to it)
```

Comment:
javadoc will squash the whitespace here; i recommend putting it in a list
```suggestion
 * <ul>
 * <li>a file
 * <li>a string (raw command line arg); or
 * <li>stdin (so we can pipe to it)
 * </ul>
```